primitiveFTPd/src/org/primftpd/filesystem/FsSshFile.java

```diff
@@ -49,6 +49,11 @@ public SshFile getParentFile() {
 		return new FsSshFile(file.getParentFile(), pftpdService, session);
 	}
 
+	@Override
+	public boolean isReadable() {
+		return super.isReadable() || isDirectory() && isExecutable();
```

Comment:
I think this should be done for other implementations as well, either in FsFile or AbstractFile

Reply:
Yeah, that's the reason it is still draft, I didn't like it either, but it was 2 AM, I had to go sleep.

Yes, in this case moving it to up (eg. FsFile, AbstractFile) makes sense, isReadable is also part of plain ftp. 👍

And the other reason for the draft: I want to do this only on the folders near to the root, so anything below eg. .../emulated/0 should be as it comes from the OS, lets send back some SSH_FX_PERMISSION_DENIED if somebody browses something in the /system (entered manually), and do this tricks only on the near-root folders, where Android also makes tricks. In these cases we will send back an empty folder list, not a SSH_FX_PERMISSION_DENIED, but it is quite expected around these near-root folders.